test/index.bs

```diff
@@ -203,27 +208,38 @@ When invoked, the {{USBTest/reset()}} method MUST return a new {{Promise}}
 |promise| and run the following steps <a>in parallel</a>:
 
 1.  Let |test| be the {{USBTest}} instance on which this method was invoked.
-1.  Set |test|.{{USBTest/chosenDevice}} to <code>null</code>.
 1.  For each {{FakeUSBDevice}} |fakeDevice| previously returned by
     {{USBTest/addFakeDevice()}}, invoke
     |fakeDevice|.{{FakeUSBDevice/disconnect()}}.
 1.  <a>Resolve</a> |promise|.
 
+When invoked, the {{USBDeviceRequestEvent/respondWith(result)}} method MUST run
+the following steps <a>in parallel</a>:
+
+1.  Wait until |result| settles.
+1.  Let |device| be <code>undefined</code>.
+1.  If |result| resolved with |response| and |response| is an instance of
+    {{FakeUSBDevice}} let |device| be |response|.
+1.  Resume execution of the suspended steps defined for {{USB/requestDevice()}}.
+1.  If |device| is <code>undefined</code> behave as if the user denied the
```

Comment:
These should kinda be parameters to the "resume execution" step, since they don't actually happen after that step finishes. You might instead do them before resuming, and say that the next use of [prompt the user to choose](https://w3c.github.io/permissions/#prompt-the-user-to-choose) should return |device| or "denied" as appropriate?